staker/bold/bold_staker.go

```diff
@@ -260,6 +260,13 @@ func (b *BOLDStaker) Start(ctxIn context.Context) {
 		}
 
 		if confirmedGlobalState != nil {
+			if agreedGlobalState == nil {
```

Comment:
I can live with that.. but seems better to read confirmedState before agreedState, and to set agreedGlobalState = confirmedGlobalState if it's nil. That way theres's no need to duplicate logic, and it makes sense (that any node sees confirmed as something it agreed to)